test-utils/derive/Cargo.toml

```diff
@@ -0,0 +1,30 @@
+[package]
+name = "substrate-test-utils-derive"
```

Comment:
name is wayy too long, maybe just test-utils? honestly could also live in a crate outside substrate.

Reply:
I named it that way for consistency with the other crates in that directory. Plus there is already a substrate-test-utils

Reply:
yeah but it's not really substrate-specific is my point

Reply:
I'm looking at the code and it looks like a `tokio-timed-test` to me :eyes:

Reply:
well... :sweat_smile: I guess!

The main reason I have this timeout is because in substrate you could start a node in a test and wait for blocks being produced. But I guess this could be one implementation for a more generic case which is "test timeout".

I did look for a crate that does exactly that but I couldn't find any. That is why I ended up doing it. Maybe I should make a PR to tokio instead.

Reply:
AH YES!!! I know why I didn't put it for tokio. There is a task executor that is proper to substrate